### ISSUE TITLE
New rule: no-settled-after-test-helper

### ISSUE THREAD

**Issue by Turbo87** [ember-cli/eslint-plugin-ember#984]
This rule should lint against using the `settled()` test helper directly after test helpers that already call `settled()` internally. Having to use this pattern is usually a sign of a deeper root cause issue that should be fixed.

**Bad**

```js
await click('.foo');
await settled();
```

```js
await fillIn('.foo');
await settled();
```

**Good**

```js
await waitFor('.foo');
await settled();
```

/cc @rwjblue

**Comment by hergaiety** [comment]
I _love_ this idea. I see this quite often, many times it's muscle memory perhaps left over from a less stable test suite or older Ember version (who knows). I often write on a PR that awaiting for settled is not needed in these cases.

**Comment by bendemboski** [comment]
Love this idea! I can't count the number of times someone has made a code change that introduced a test-race-condition and instead of digging in to find what needed to be test-waited, threw in an extra await settled(), leaving a ticking random-test-failure time bomb.

**Comment by rwjblue** [comment]
We should also catch:

```js
await settled();
await settled();
```
